process.go

```diff
@@ -79,39 +88,50 @@ func (p *process) Kill(ctx context.Context, s syscall.Signal) error {
 	return err
 }
 
-func (p *process) Wait(ctx context.Context) (uint32, error) {
+func (p *process) Wait(ctx context.Context) (<-chan ExitCode, error) {
 	cancellable, cancel := context.WithCancel(ctx)
 	defer cancel()
 	eventstream, err := p.task.client.EventService().Subscribe(cancellable, &eventsapi.SubscribeRequest{
 		Filters: []string{"topic==" + runtime.TaskExitEventTopic},
 	})
 	if err != nil {
-		return UnknownExitStatus, err
+		return nil, err
 	}
 	// first check if the task has exited
 	status, err := p.Status(ctx)
 	if err != nil {
-		return UnknownExitStatus, errdefs.FromGRPC(err)
+		return nil, errdefs.FromGRPC(err)
 	}
+
+	chStatus := make(chan ExitCode, 1)
 	if status.Status == Stopped {
-		return status.ExitStatus, nil
+		chStatus <- ExitCode{Code: status.ExitStatus}
+		return chStatus, nil
 	}
-	for {
-		evt, err := eventstream.Recv()
-		if err != nil {
-			return UnknownExitStatus, err
-		}
-		if typeurl.Is(evt.Event, &eventsapi.TaskExit{}) {
-			v, err := typeurl.UnmarshalAny(evt.Event)
+
+	go func() {
+		for {
```

Comment:
to be extra sure, maybe send a dummy entry here and sync/discard it before returning?